### PR TITLE
Models refactor

### PR DESCRIPTION
**Pull request by taitfuller**
Refactor models to add state to Round and additional initialisation and validation tests

### CODE REVIEW

File: backend/src/models/game.model.ts [hunk 1]
```diff
@@ -6,23 +6,20 @@ import { RoundSchema, Round } from "./round.model";
 
 export enum GameState {
   lobby = "LOBBY",
-  roundBefore = "ROUND_BEFORE",
-  roundPlayers = "ROUND_PLAYERS_CHOOSE",
-  roundHost = "ROUND_HOST_CHOOSE",
-  roundAfter = "ROUND_AFTER",
+  active = "ACTIVE",
```
Comment (Alxjlv) [inline backend/src/models/game.model.ts:9]: Ah I see why the refactor was needed lol

File: backend/src/models/game.model.ts [hunk 2]
```diff
@@ -6,23 +6,20 @@ import { RoundSchema, Round } from "./round.model";
...
-  discardedSetups?: Types.DocumentArray<Setup>;
+  discardedSetups: Types.DocumentArray<Setup>;
   punchlines: string[];
-  discardedPunchlines?: string[];
+  discardedPunchlines: string[];
   players: Types.DocumentArray<Player>;
-  state?: GameState;
-  rounds?: Types.DocumentArray<Round>;
+  state: GameState;
+  rounds: Types.DocumentArray<Round>;
```
Comment (Alxjlv) [inline backend/src/models/game.model.ts:22]: Why are you getting rid of them being optional? Was it an oversight prior?

File: backend/src/models/round.model.ts [hunk 3]
```diff
@@ -1,12 +1,28 @@
-import { Document, ObjectId, Schema } from "mongoose";
+import { Document, ObjectId, Schema, Types } from "mongoose";
 import { SetupSchema, Setup } from "./setup.model";
+import { Player } from "./player.model";
+
+export enum RoundState {
+  before = "BEFORE",
+  playersChoose = "PLAYERS_CHOOSE",
+  hostChooses = "HOST_CHOOSES",
```
Comment (Alxjlv) [inline backend/src/models/round.model.ts:8]: It's not the host choosing right? Unless you're just calling The Man™️ host for ease of use

**Review by Alxjlv — COMMENTED** [review]
I'm a little bit confused by round - I thought the host didn't play any significance once the game begins? Are you using host in the place of The Man™️ or is the host still technically used in the backend?

File: backend/src/models/game.model.ts (hunk 1 above)
Comment (taitfuller) [reply backend/src/models/game.model.ts:9]: Yeah lol just simplifying the state management 👌🏼

File: backend/src/models/game.model.ts (hunk 2 above)
Comment (taitfuller) [reply backend/src/models/game.model.ts:22]: Yeah this was a misunderstanding on how the Mongoose model typing works - optional in the interface means that field may be undefined in the DB which is not what we want - we just want some fields to be required to be set to initialise (such as gameCode, punchlines, setups) & others that default to an initial value

File: backend/src/models/round.model.ts (hunk 3 above)
Comment (taitfuller) [reply backend/src/models/round.model.ts:8]: Ah right so for all intents & purposes "the host" is "The Man™" in the backend so we don't have to manage extra state

Comment (Alxjlv) [reply backend/src/models/round.model.ts:8]: Right so the host role is shifted around in the backend? Does this mean that the person who started the original lobby is the first The Man™️?

**Review by Alxjlv — APPROVED** [review]
Looks solid to me!

**Pull request merged**